test/OpenTelemetry.Tests/Logs/LogRecordTest.cs

```diff
@@ -107,6 +114,65 @@ public void CheckStateForUnstructuredLogWithStringInterpolation()
             Assert.Equal(message.ToString(), state.ToString());
         }
 
+        [Fact]
+        public void StateCanBeSetByProcessor()
+        {
+            using var loggerFactory = LoggerFactory.Create(builder => builder
+                .AddOpenTelemetry(options =>
+                {
+                    options.AddProcessor(new RedactionProcessor(Field.State));
+                    options.AddInMemoryExporter(this.exportedItems);
+                }));
+
+            var logger = loggerFactory.CreateLogger<LogRecordTest>();
+            {
+                logger.LogInformation($"This should be replaced {0.99}.");
+
+                var state = this.exportedItems[0].State as IReadOnlyList<KeyValuePair<string, object>>;
```

Comment:
if the goal is to test that State can be set, then we dont need ReadOnlyList.... Simply assign State to "newstate", and validate it here.

Reply:
I'm not sure how to do it, because it seems there is no proper .ToString() overload for the State as an object.